tools/pylint-extensions/azure-pylint-guidelines-checker/tests/test_pylint_custom_plugins.py

```diff
@@ -4829,4 +4829,121 @@ def test_allowed_import_else(self):
             self.checker.visit_importfrom(ima)
             self.checker.visit_import(imb)
             self.checker.visit_import(imc)
-            self.checker.visit_importfrom(imd)
\ No newline at end of file
+            self.checker.visit_importfrom(imd)
+
+
+class TestDoNotLogErrorsEndUpRaising(pylint.testutils.CheckerTestCase):
+
+    """Test that any errors raised are not logged at 'error' or 'warning' levels in the exception block."""
+
+    CHECKER_CLASS = checker.DoNotLogErrorsEndUpRaising
+
+    def test_error_level_not_logged(self):
+        """Check that any exceptions raised aren't logged at error level in the exception block."""
+        exception_node = astroid.extract_node('''
+        try:
+            add = 1 + 2
+        except Exception as e:
+            logger.error(str(e))
```

Comment:
The logic here is actually going to be trickier than this I think. Because it's quite possible to have lots of varying branches in an `except` clause. Some of our error handling cases have pretty complex code. For example:
```python
except SomeException as e:
    if e.code == "Retryable":
        logger.warning(f"Retriable failure occured: {e}, attempting to restart")  # This kind of logging is okay
        return True
    elif isinstance(SomeException(e, SomeDerivedException):
        logger.error(f"System shutting down due to error: {e}.")  # This logging is also okay I think
        return False
    logger.error(f"Unexpected error occurred: {e}") # This is the logging we'd want to flag as excessive.
    raise SystemError("Uh oh") from e
```
I'm not even sure if it's possible to make such a fine-grained distinction - as I'm not sure whether we can parse branches. If we can parse branches, then we'd probably want to flag branches that end in either:
- `raise`
- `raise <ExceptionType> from <exception_instance>`

Let's add some more complex test scenarios here to see how far we can push this. But if it's not possible with the data we have then we can get by.

Reply:
I have added new code to parse branches and sub-branches. Have re-run against the SDKs and found several files that don't comply with the new rules, I can attach these if you'd like. No false positives from testing against SDKs.